palette/src/rgb/rgb.rs

```diff
@@ -999,6 +1009,27 @@ impl<S> FromStr for Rgb<S, u8> {
     }
 }
 
+impl<S> FromStr for Rgba<S, u8> {
+    type Err = FromHexError;
+
+    /// Parses a color hex code of format '#ff00bbff' (with or without the leading '#') into a
+    /// [`Rgba<S, u8>`] instance.
+    fn from_str(hex: &str) -> Result<Self, Self::Err> {
+        let hex_code = hex.strip_prefix('#').map_or(hex, |stripped| stripped);
+        match hex_code.len() {
+            8 => {
+                let red = u8::from_str_radix(&hex_code[..2], 16)?;
+                let green = u8::from_str_radix(&hex_code[2..4], 16)?;
+                let blue = u8::from_str_radix(&hex_code[4..6], 16)?;
+                let alpha = u8::from_str_radix(&hex_code[6..8], 16)?;
+                let col: Rgba<S, u8> = Rgba::new(red, green, blue, alpha);
+                Ok(col)
+            }
+            _ => Err("invalid rgba hex code format".into()),
```

Comment:
We know this should become an rgba hex error, so we could make one explicitly instead of using `into`.